tests/test_other.py

```diff
@@ -735,17 +745,23 @@ def test_incorrect_c_detection(self):
 
   def test_odd_suffixes(self):
     for suffix in ['CPP', 'c++', 'C++', 'cxx', 'CXX', 'cc', 'CC', 'i', 'ii']:
+      if self.is_wasm_backend() and suffix == 'ii':
+        # wasm backend treats .i and .ii specially and considers them already
+        # pre-processed.  Because if this is strips all the -D command line
+        # flags, including the __EMSCRIPTEN__ define, which makes this fail
+        # to compile since libcxx/__config depends in __EMSCRIPTEN__.
+        continue
```

Comment:
should we change the asm.js backend behavior? seems an odd difference to have. maybe not that important though (definitely not in this PR)

Reply:
It does seem like an odd thing.  I didn't even know about .ii files before!

I'm not sure how its working with fastcmp right now TBH.. I think maybe its becasue with fastcmp the EMSCRIPTEN define comes from the compiler and not from the command line?  For consistency we might want to fix that, since we always add it in with -D anyway.

Perhaps we should modify this test such to make sure that '-D' flags are ignored with 'i' and 'ii' files?  Most likely a followup PR.

Reply:
Sounds good for a followup.